### PR TITLE
RK-6817 - Adding server selection screen documentation

### PR DESCRIPTION
**Pull request by noagoldman**

### CODE REVIEW

File: docs/server-selection.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+---
+id: servers-selection
```
Comment (ElDuderinos) [inline docs/server-selection.md:2]: Maybe keep consistency with the filename?
"servers-selection" / "server-selection.md" ?

File: docs/server-selection.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+
+Now, you can see the instances you will debug on the next debug session,\
+chosen according to your filters. Click **Let's go** and start your debug session.
+ 
+**Bookmarks** \
+You may want to save the state of your debug session. It can be done by using a bookmark.
+It will save the following:
+1. Connected source code repositories.
+2. One or set of filters that have been selected.
```
Comment (ElDuderinos) [inline docs/server-selection.md:31]: I think it should be "One or a set of filters that have been selected" or "One or the set of filters that have been selected".

**Review by ElDuderinos — CHANGES_REQUESTED** [review]

File: docs/server-selection.md (hunk 2 above)
Comment (noagoldman) [reply docs/server-selection.md:31]: Yep you are right

File: docs/server-selection.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+---
+The Server Selection screen will allow you to define the context of your debug session.
+
+### Set Up your Debug Session 
+To start a debug session, define at least one of the following filters:
+
+<img src="/img/screenshots/server_selection_screen.png"/>
+
+**Env**:\
+Choose instances that are running in a specific environment.
```
Comment (niryarden) [inline docs/server-selection.md:14]: don't like the fact we're using `servers` and `instances` intermittently. shall we choose one term and stick to it?

File: docs/server-selection.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+**Env**:\
+Choose instances that are running in a specific environment. 
+
+**Repository**:\
+Choose instances according to their source code repositories.
+
+**Other Filters**:\
+Filter your instances by [labels](projects-labels.md) that were defined during the Rookout SDK installation. 
+Choose instances with a specific hostname, labels, process, IP, etc…
```
Comment (niryarden) [inline docs/server-selection.md:21]: I think we should add `or` (or any similar word that sounds better) between the two sentences. otherwise, it sounds like they have to do both.

File: docs/server-selection.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+<img src="/img/screenshots/server_selection_screen.png"/>
+
+**Env**:\
+Choose instances that are running in a specific environment. 
+
+**Repository**:\
+Choose instances according to their source code repositories.
+
+**Other Filters**:\
+Filter your instances by [labels](projects-labels.md) that were defined during the Rookout SDK installation.
```
Comment (niryarden) [inline docs/server-selection.md:20]: I'd consider saying `manually defined`, to emphasize it is their to decide 
Additionally, it's not really during the sdk **_installation_**, more like during the SDK **_setup_**.

File: docs/server-selection.md [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+Now, you can see the instances you will debug on the next debug session,\
+chosen according to your filters. Click **Let's go** and start your debug session.
+ 
+**Bookmarks** \
+You may want to save the state of your debug session. It can be done by using a bookmark.
+It will save the following:
+1. Connected source code repositories.
+2. One or a set of filters that have been selected.
+
```
Comment (niryarden) [inline docs/server-selection.md:32]: two things I think we should add:
1. A screenshot of the pencil in the top of the left pane. saying something like: 'click on the pencil to go to the server selection page and edit your debug session'. 
2. A paragraph about 'making your server selection page better' with an explanation that choose env is based on labels with env key (we must say it somewhere in the docs, here looks like a good place, and then i'll be able to reference to it in the docs link app itself). And also explain that the repo selection is based on the git_origin definition of your sdk setup, with a ref to the existing auto source fetching section in the docs.

**Review by niryarden — CHANGES_REQUESTED** [review]
Everything is only my opinion. and I'm far from being a docs expert (or an English writing expert). So consult further with @Rookeret73 / @yonitwis if you're not sure about anything I commented.

File: docs/server-selection.md (hunk 6 above)
Comment (niryarden) [reply docs/server-selection.md:32]: if you indeed add this paragraph, you can also mention in adding labels to your sdk setup. with a link to the labels page in the docs.
then, you'll be able to shorten the other filters paragraph without clearly distinguishing between labels and the rest of the filters, just saying something like `filter the servers using their labels, hostname, labels, process, IP, etc…`

**Comment by niryarden** [comment]
also, we might want to completely delete the dynamic projects page. not only top remove it from the contents on the left pane.
so it won't be reachable by searching

File: docs/server-selection.md [hunk 7]
```diff
@@ -0,0 +1,36 @@
+---
+id: server-selection
+title: Debug session setup
+sidebar_label: Servers Selection
+---
+This screen will allow you to define the context of your debug session.
+
+### Set Up your Debug Session 
+To start a debug session, define at least one of the following filters:
```
Comment (niryarden) [inline docs/server-selection.md:9]: not sure about the word 'define'. maybe 'select'?
because they're selecting from a list of options, not inventing filters from the head.
'select' is also the word we use in the UI.

File: docs/server-selection.md [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+
+**Env**:\
+Choose instances that are running in a specific environment. 
+
+**Repository**:\
+Choose instances according to their source code repositories.
+
+**Other Filters**:\
+Filter your instances by [labels](projects-labels.md) that were manually defined during the Rookout SDK setup, 
+Filter by a specific hostname, label, process, IP, etc…
```
Comment (niryarden) [inline docs/server-selection.md:21]: `the Rookout SDK setup, or filter`
(add or, filter should not be capital case.)

File: docs/server-selection.md [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+
+
+Now, you can see the instances you will debug on the next debug session,\
+chosen according to your filters. Click **Let's go** and start your debug session.
```
Comment (niryarden) [inline docs/server-selection.md:25]: personally, i think `on the next debug session, according to the selected filters` is a bit clearer sentence.
and also maybe `Click let's go to start your debug session`.

both not a must

File: docs/server-selection.md [hunk 10]
```diff
@@ -0,0 +1,36 @@
+---
+id: server-selection
+title: Debug session setup
+sidebar_label: Servers Selection
```
Comment (niryarden) [inline docs/server-selection.md:4]: sidebar label is still server selection...

File: website/sidebars.json [hunk 11]
```diff
@@ -13,7 +13,7 @@
       "deployment-examples"
     ],
     "Debug Session": [
-      "projects",
+      "server-selection",
```
Comment (niryarden) [inline website/sidebars.json:16]: debug session setup

File: docs/server-selection.md [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+Filter your instances by [labels](projects-labels.md) that were manually defined during the Rookout SDK setup, 
+Filter by a specific hostname, label, process, IP, etc…
+
+
+Now, you can see the instances you will debug on the next debug session,\
+chosen according to your filters. Click **Let's go** and start your debug session.
+ 
+
+<img src="/img/screenshots/pencil_to_server_selection.png"/>
+Click on the pencil to **edit your debug session**.
```
Comment (niryarden) [inline docs/server-selection.md:29]: maybe we should give some context?

`Later, while working inside your debug session, you can click the pencil any time to edit it again`.
wdyt?

File: docs/server-selection.md [hunk 13]
```diff
@@ -0,0 +1,36 @@
...
+chosen according to your filters. Click **Let's go** and start your debug session.
+ 
+
+<img src="/img/screenshots/pencil_to_server_selection.png"/>
+Click on the pencil to **edit your debug session**. 
+
+**Bookmarks** \
+You may want to save the state of your debug session. It can be done by using a bookmark.
+It will save the following:
+1. Connected source code repositories.
```
Comment (niryarden) [inline docs/server-selection.md:34]: we should reconsider that statement IMO.
the whole bookmarks feature changed, and just saying it saves connected sources is not very correct:
1. when you save a bookmark you can't even see what are your connected sources. it's transparent to you in this stage.
2. manually added sources aren't removed from the debug session if you switch bookmarks. they stay there until you manually remove them.
3. auto-added sources would be auto-fetched following the bookmark filters loading anyway, so the fact that the bookmark saves them is practically pointless.

In general, I think we can consider deprecating the bookmark feature or rethink the way it works.
but specifically regarding this PR - I wouldn't mention the sources. I would rephrase this whole paragraph, explaining that bookmarks are made for naming different debug session setups, by saving sets of selected filters.

**Review by niryarden — CHANGES_REQUESTED** [review]
hey, sorry if I'm giving you hard time with this PR :(
two general comments:
1. I think we should remove from this pr any reference to the server-selection name. it's a name we used for the design and development of the feature, and it's not relevant anymore. I believe future editors of the docs will be able to find the right spot, and that the uniformity we'll get by sticking to the costumer-facing name will make everything easier and prevent confusion.
2. I do believe that this page should include an explanation about the env dropdown and how it works, because I believe this is the place where users will look for it. That's my opinion - but it's your call.

File: docs/server-selection.md (hunk 12 above)
Comment (noagoldman) [reply docs/server-selection.md:29]: How about: Click on the pencil from your / the current debug session to edit and change instances?

Comment (niryarden) [reply docs/server-selection.md:29]: yeah whatever sounds good to you.
just wanted it to be in the context of being inside a debug session

File: docs/server-selection.md (hunk 13 above)
Comment (noagoldman) [reply docs/server-selection.md:34]: I don't think we want to deprecate the bookmark feature, users use it a lot. I will remove it from this page though :)

File: docs/server-selection.md (hunk 12 above)
Comment (noagoldman) [reply docs/server-selection.md:29]: Understood

File: docs/server-selection.md (hunk 13 above)
Comment (niryarden) [reply docs/server-selection.md:34]: the comment about deprecating is just a thought i had, but i agree we should think about it further.

regarding the docs - i wouldn't remove it completely, because this is where users will look for an answer when they'll see it in the server selection page and won't understand what it is.
I do think we should mention it, but I think we should not mention source repos are involved

**Review by niryarden — APPROVED** [review]
